applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -406,14 +406,40 @@ namespace Kratos
         const ProcessInfo& rCurrentProcessInfo 
         )
     {
-        if ( rOutput.size() != GetGeometry().IntegrationPoints(  ).size() )
+        const GeometryType::IntegrationPointsArrayType& integration_points = GetGeometry().IntegrationPoints();
+        if ( rOutput.size() != integration_points.size() )
         {
-            rOutput.resize( GetGeometry().IntegrationPoints(  ).size() );
+            rOutput.resize( integration_points.size() );
         }
 
-        for ( unsigned int ii = 0; ii < mConstitutiveLawVector.size(); ii++ )
+        if (rVariable == GAUSS_WEIGHT)
         {
-            rOutput[ii] = mConstitutiveLawVector[ii]->GetValue( rVariable, rOutput[ii] );
+            const unsigned int number_of_nodes = GetGeometry().size();
+            const unsigned int dim = GetGeometry().WorkingSpaceDimension();
+            //const unsigned int strain_size = GetProperties().GetValue( CONSTITUTIVE_LAW )->GetStrainSize();
+
+            Matrix DN_DX( number_of_nodes, dim );
+            Matrix J0(dim, dim);
+            Matrix InvJ0(dim, dim);
+
+            for ( unsigned int point_number = 0; point_number < integration_points.size(); point_number++ )
```

Comment:
I wil reccomend to use the functions I implemented the last days CalculateKinematicVariables()

This will reduce the code duplications and wiil be more generic (in case we implement UL elements, or taking into consideration the fact that in the kinematic_linear_element the detJ is calculated in a different way)